flink-python/pyflink/datastream/tests/test_stream_execution_environment.py

```diff
@@ -211,3 +214,31 @@ def test_execute(self):
         self.assertEqual(len(execution_result.get_all_accumulator_results()), 0)
         self.assertIsNone(execution_result.get_accumulator_result('accumulator'))
         self.assertIsNotNone(str(execution_result))
+
+    def test_from_collection_without_data_types(self):
+        ds = self.env.from_collection([(1, 'Hi', 'Hello'), (2, 'Hello', 'Hi')])
+        test_sink = DataStreamTestCollectSink(True)
+        ds._j_data_stream.addSink(test_sink._j_data_stream_test_collect_sink)
```

Comment:
There are some problems with the design:
1. we need to pass java datastream object explicitly which should be avoided.
2. we need to specify True false manually. I think we can infer the true false from the input datastream, i.e., set true when output_type is `PickledBytesTypeInfo`.

The code may looks like:
```
collect_util = DataStreamCollectUtil()
collect_util.collect(ds)
results = collect_util.results()
```

Reply:
Yes, there might be some spaces for improvements.